src/pybind/mgr/dashboard/frontend/src/app/shared/api/host.service.ts

```diff
@@ -141,7 +141,9 @@ export class HostService extends ApiClient {
         const devices = _.flatMap(hosts, (host) => {
           return host.devices.map((device) => {
             device.hostname = host.name;
-            device.uid = device.device_id ? device.device_id : `${device.hostname}-${device.path}`;
+            device.uid = device.device_id
+              ? `${device.device_id}-${device.hostname}-${device.path}`
+              : `${device.hostname}-${device.path}`;
```

Comment:
I'd like some tests here. If `device_id` is an empty string, why not just simplify to:
```suggestion
            device.uid = `${device.device_id}-${device.hostname}-${device.path}`;
```

Reply:
I'll add tests here. 
but with your approach if device_id is empty, won't the uid start with a `-`.